Extend/Payment_Gateway.php

```diff
@@ -187,92 +202,177 @@ public function makeTransaction($dAmount, $oOrder)
-            $oTransaction->setBasket($wdBasket);
+    /**
+     *
+     * build an account holder
+     *
+     * @param Order $oOrder
+     * @param User $oUser
+     * @return AccountHolder
+     */
+    private function _buildAccountHolder(Order $oOrder, User $oUser): AccountHolder
+    {
+        $oAccountHolder = new \Wirecard\PaymentSdk\Entity\AccountHolder();
+
+        $oAddress = new \Wirecard\PaymentSdk\Entity\Address(
+            self::_getCountryCode($oUser->oxuser__oxcountryid),
+            $oUser->oxuser__oxcity->value,
+            $oUser->oxuser__oxstreet->value . ' ' . $oUser->oxuser__oxstreetnr->value
+        );
+
+        $oAddress->setPostalCode($oUser->oxuser__oxzip->value);
+
+        $sState = $oUser->getStateTitle();
+        if (!empty($sState)) {
+            $oAddress->setState($sState);
         }
-        $oTransaction->setNotificationUrl($sShopUrl . 'notify.php');
-        $oResponse = $oTransactionService->pay($oTransaction);
-        return $oResponse;
+
+        $sStreet2 = $oUser->oxuser__oxaddinfo->value;
+        if (!empty($sStreet2)) {
+            $oAddress->setStreet2($sStreet2);
+        }
+
+        $oAccountHolder->setAddress($oAddress);
+        $oAccountHolder->setFirstName($oUser->oxuser__oxfname->value);
+        $oAccountHolder->setLastName($oUser->oxuser__oxlname->value);
+        $oAccountHolder->setPhone($oUser->oxuser__oxfon->value);
+        $oAccountHolder->setEmail($oOrder->oxorder__oxbillemail->value);
+
+        return $oAccountHolder;
+    }
+
+    /**
+     *
+     * build the shipping account holder
+     *
+     * @param Order $oOrder
+     * @param User $oUser
+     * @return AccountHolder
+     */
+    private function _buildShipping(Order $oOrder, User $oUser): AccountHolder
+    {
+        $oAddress = new \Wirecard\PaymentSdk\Entity\Address(
+            self::_getCountryCode($oOrder->oxorder__oxdelcountryid),
+            $oOrder->oxorder__oxdelcity->value,
+            $oOrder->oxorder__oxdelstreet->value . ' ' . $oUser->oxuser__oxdelstreetnr->value
+        );
+        $oAddress->setPostalCode($oOrder->oxorder__oxdelzip->value);
+
+        $sStateId = $oOrder->oxOrder__oxdelstateid->value;
+        if (!empty($sStateId)) {
+            $oState = oxNew(State::class);
+            $oAddress->setState($oState->getTitleById($sStateId));
+        }
+
+        $sStreet2 = $oOrder->oxOrder__oxdeladdinfo->value;
+        if (!empty($sStreet2)) {
+            $oAddress->setStreet2($sStreet2);
+        }
+
+        $oAccount = new \Wirecard\PaymentSdk\Entity\AccountHolder();
+        $oAccount->setAddress($oAddress);
+        $oAccount->setFirstName($oOrder->oxorder__oxdelfname->value);
+        $oAccount->setLastName($oOrder->oxorder__oxdellname->value);
+        $oAccount->setPhone($oOrder->oxorder__oxdelfon->value);
+
+        return $oAccount;
+    }
+
+    private function _addBasketInfo(Transaction &$oTransaction, Basket $oBasket)
+    {
+        $finalPrice = array();
```

Comment:
In my opinion, the variable name is a bit misleading because it is not a single value, but rather an array of multiple **unit prices**.